File: 0061-rotate-list/0061-rotate-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* rotateRight(ListNode* head, int k) {
        if(!head || !head->next) return head;
        int len=0;
        ListNode* node=head,*prev=NULL;
        while(node){
            prev=node;
            node=node->next;
            len++;
        }

        int cnt=len-(k%len);
        node=head;
        if(cnt==len) return head;
        while(node && --cnt){
            node=node->next;
        }
        ListNode *newhead=node->next;
        node->next=NULL;
        prev->next=head;
        return newhead;
    }
};

File: 0061-rotate-list/0061-rotate-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def rotateRight(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:
        if not(head): 
            return head
        cnt = 0 
        node = head 
        while node: 
            cnt+=1 
            node = node.next 

        k= k%cnt if k>=cnt else k 
        if k==0: 
            return head 
        l1,l2=head,head
    
        while l2.next : 
            k-=1
            if k<0: 
                l1=l1.next 
            l2=l2.next 

        l2.next = head 
        newhead = l1.next 
        l1.next = None 
        return newhead 
    